tests/test_pulse_number.py

```diff
@@ -64,6 +64,12 @@ def test_pulse_number(model, toas):
     assert "pulse_number" in toas.table.colnames
 
 
+def test_remove_pulse_number(model, toas):
```

Comment:
It'd be good to have a test to demonstrate what happens when pulse numbers aren't present and are removed - this is where one specifies whether or not there should be a warning.

Reply:
ok, added